### PR TITLE
Mccalluc/link to lineup

### LINKED ISSUE #2278: Implement LineUp link in search page
See design in #2271 

Lineup Links:
Donors: https://portal.hubmapconsortium.org/lineup/donors
Samples: https://portal.hubmapconsortium.org/lineup/samples
Datasets: https://portal.hubmapconsortium.org/lineup/datasets

### PR DESCRIPTION
**Pull request by mccalluc**
Fix #2278

Began by renaming `DownloadButton` to `MetadataMenu`, so not everything here is actually new.

Still a draft, but some questions, @john-conroy :
- Is there a better way to get the menu positioned below the button than the `PaperProps`?
- For the `InfoIcon` would it be worth introducing a reusable small version? Or do the different contexts where it is used want more flexibility?
- Right now the "Visualize" and "Download" are styled similarly, though the "Visualize" is underlined on hover... Is this a good way to do it? Should I override the default hover style?

### CODE REVIEW

File: context/app/static/js/components/searchPage/MetadataMenu/MetadataMenu.jsx [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+  const allResultsUUIDs = useSearchViewStore((state) => state.allResultsUUIDs);
+
+  const [anchorEl, setAnchorEl] = React.useState(null);
+  const open = Boolean(anchorEl);
+  const handleClick = (event) => {
+    setAnchorEl(event.currentTarget);
+  };
+  const handleClose = () => {
+    setAnchorEl(null);
+  };
```
Comment (john-conroy) [inline context/app/static/js/components/searchPage/MetadataMenu/MetadataMenu.jsx:25]: Use separate state for open/closed and don't manage the ref.

**Comment by john-conroy** [comment]
> Right now the "Visualize" and "Download" are styled similarly, though the "Visualize" is underlined on hover... Is this a good way to do it? Should I override the default hover style?

I would override the style `text-decoration: none;` should do the trick.

**Comment by mccalluc** [comment]
@john-conroy -- switching over to the vitessce problem now. After standup, if you can help me sort out how the refs should work here with the new approach.

File: context/app/static/js/components/searchPage/MetadataMenu/MetadataMenu.jsx [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+import MenuItem from '@material-ui/core/MenuItem';
+
+import { InfoIcon } from 'js/shared-styles/icons';
+import useSearchViewStore from 'js/stores/useSearchViewStore';
+import { createDownloadUrl } from 'js/helpers/functions';
+import { SecondaryBackgroundTooltip } from 'js/shared-styles/tooltips';
+import { StyledButton, StyledLink } from './style';
+
+function useMenu(initialState) {
+  // TODO: Pull out into utilities.
```
Comment (john-conroy) [inline context/app/static/js/components/searchPage/MetadataMenu/MetadataMenu.jsx:16]:
```suggestion
  // TODO: Pull out into hooks.
```

**Comment by mccalluc** [comment]
@john-conroy -- Do you want to finish this off and merge, or should I?

**Comment by john-conroy** [comment]
I think we agreed yesterday I would, thanks!

**Comment by john-conroy** [comment]
Changes made in #2355. Merging.

**Pull request merged**